data/mods/Magiclysm/worldgen/elf_workshop.json

```diff
@@ -0,0 +1,102 @@
+[
+  {
+    "type": "palette",
+    "id": "elf_palette",
+    "terrain": { "3": "t_rdoor_c", "1": "t_strconc_wall", ".": "t_metal_floor_olight" },
+    "furniture": {
+      "a": "f_anvil",
+      "t": "f_table",
+      "b": "f_tablesaw",
+      "c": "f_chair",
+      "d": "f_drill_press",
+      "j": "f_jointer",
+      "r": "f_router",
+      "w": "f_workbench",
+      "m": "f_workbench",
+      "s": "f_stool",
+      "f": "f_demon_forge"
+    },
+    "items": {
+      "a": [
+        { "item": "tools_anvil", "chance": 75, "repeat": [ 1, 2 ] },
+        { "item": "mithril_ingot", "chance": 10 },
+        { "item": "scrap_copper", "chance": 40 }
+      ],
+      "m": [
+        { "item": "tools_anvil", "chance": 25 },
+        { "item": "clamp", "chance": 50, "repeat": [ 1, 3 ] },
+        { "item": "metal_file", "chance": 50 },
+        { "item": "pin_reamer", "chance": 25 },
+        { "item": "mithril_ingot", "chance": 10 },
+        { "item": "scrap_copper", "chance": 80, "repeat": [ 1, 3 ] }
+      ],
+      "w": [
+        { "item": "misc_repairkit", "chance": 25 },
+        { "item": "hammer", "chance": 90 },
+        { "item": "sandpaper", "chance": 90 },
+        { "item": "clamp", "chance": 90, "repeat": [ 2, 4 ] },
+        { "item": "xacto", "chance": 10 }
+      ],
+      "f": [ { "item": "alumentum" } ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": "elf_workshop",
+    "object": {
+      "fill_ter": "t_metal_floor",
+      "rows": [
+        "111111111111111111111111",
+        "1fa.....fa..........jsw1",
+        "1ms......m....d....r.ct1",
+        "1d...................ct1",
+        "1bsw.........c........d1",
+        "1dsw........ctc......af1",
+        "1............c.........3",
+        "1msa...................1",
+        "1d.f..................d1",
+        "1......................3",
+        "1bsw..................b1",
+        "1....d............j....1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1......................3",
+        "1......................1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1...........m..........1",
+        "1f.........sa.........f1",
+        "1a.sm......f.......ms.a1",
+        "111111111111111111111111"
+      ],
+      "palettes": [ "elf_palette" ],
+      "place_monsters": [
```

Comment:
That's a lot of elves.

Reply:
This still needs a roof mapgen entry as well.  Please see the [JSON_ROOF_MAPGEN.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_Mapping_Guides/JSON_ROOF_MAPGEN.md)